cmd/service_run.go

```diff
@@ -326,17 +341,48 @@ func startTokenDApp(log *zap.Logger, rf *RootFlags, openBrowser bool, cfg *netwo
 }
 
 // waitSig will wait for a sigterm or sigint interrupt.
-func waitSig(ctx context.Context, cfunc func(), log *zap.Logger) {
+func waitSig(ctx context.Context, cfunc func(), log *zap.Logger, pendingSigRequests chan service.ConsentRequest, sigRequestsResponses chan service.ConsentConfirmation, p *printer.InteractivePrinter) {
 	gracefulStop := make(chan os.Signal, 1)
+
 	signal.Notify(gracefulStop, syscall.SIGTERM)
 	signal.Notify(gracefulStop, syscall.SIGINT)
 	signal.Notify(gracefulStop, syscall.SIGQUIT)
 
-	select {
-	case sig := <-gracefulStop:
-		log.Info("caught signal", zap.String("signal", fmt.Sprintf("%+v", sig)))
-		cfunc()
-	case <-ctx.Done():
-		// nothing to do
+	for {
+		select {
+		case sig := <-gracefulStop:
+			log.Info("caught signal", zap.String("signal", fmt.Sprintf("%+v", sig)))
+			cfunc()
+			return
+		case <-ctx.Done():
+			// nothing to do
+			return
+		case ev := <-pendingSigRequests:
+			txStr, err := ev.String()
+			if err != nil {
+				log.Info("failed to marshall signature request content")
+				cfunc()
+				return
+			}
+			p.CheckMark().Text("Received TX sign request: ").WarningText(txStr).NextLine()
+			reader := bufio.NewReader(os.Stdin)
+			p.CheckMark().WarningText("Please accept or decline sign request: (y/n)").NextLine()
+			answer, err := reader.ReadString('\n')
+			answer = strings.TrimSuffix(answer, "\n")
+			if err != nil {
+				log.Error("failed to read user input", zap.Error(err))
+				cfunc()
+				return
+			}
+			if answer == "y" || answer == "Y" {
+				log.Info("user approved signature for transaction", zap.Any("transaction", txStr))
```

Comment:
Minor: its not the signature that was declined, but the signing of the transaction so "user approved sign request" is better. Similar for the rejection log.

Reply:
indeed, that's a mistranslation from me. In French signature also means the act of signing
done.